cmd/catalyst-node/catalyst-node.go

```
@@ -60,21 +73,149 @@ func runClient() error {
 	for {
 		event := <-eventCh
-		fmt.Printf("got event: %v\n", event)
-		// eli note: this is the part where we trigger reconcillation with Mist
+		fmt.Printf(" got event: %v\n", event)
+
+		members, _ := getSerfMembers(client)
+		balancedServers := getMistLoadBalancerServers(config.mistLoadBalancerEndpoint)
+
+		membersMap := make(map[string]bool)
+
+		for _, member := range members {
+			member_host := member.Addr.String()
+
+			// commented out as for now the load balancer does not return ports
+			//if member.Port != 0 {
+			//	member_host = fmt.Sprintf("%s:%d", member_host, member.Port)
+			//}
+
+			membersMap[member_host] = true
+		}
+
+		fmt.Printf("current members in cluster: %v\n", membersMap)
+		fmt.Printf("current members in load balancer: %v\n", balancedServers)
+
+		// compare membersMap and balancedServers
+		// del all servers not present in membersMap but present in balancedServers
+		// add all servers not present in balancedServers but present in membersMap
+
+		// note: untested as per MistUtilLoad ports
+		for k := range balancedServers {
+			if _, ok := membersMap[k]; !ok {
+				fmt.Printf("deleting server %s from load balancer\n", k)
+				changeLoadBalancerServers(config.mistLoadBalancerEndpoint, k, "del")
+			}
+		}
+
+		for k := range membersMap {
+			if _, ok := balancedServers[k]; !ok {
+				fmt.Printf("adding server %s to load balancer\n", k)
+				changeLoadBalancerServers(config.mistLoadBalancerEndpoint, k, "add")
+			}
+		}
+
 	}
 
 	return nil
 }
 
+func connectSerfAgent(serfRPCAddress string, serfRPCAuthKey string) (*serfclient.RPCClient, error) {
+	return serfclient.ClientFromConfig(&serfclient.Config{
+		Addr:    serfRPCAddress,
+		AuthKey: serfRPCAuthKey,
+	})
+}
+
+func getSerfMembers(client *serfclient.RPCClient) ([]serfclient.Member, error) {
+	return client.Members()
+}
+
+func changeLoadBalancerServers(endpoint string, server string, action string) {
```

Should return an error